cats/src/main/scala/com/gu/scanamo/ScanamoCats.scala

```diff
@@ -0,0 +1,84 @@
+package com.gu.scanamo
+
+import cats.effect.Effect
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBAsync
+import com.amazonaws.services.dynamodbv2.model.{BatchWriteItemResult, DeleteItemResult}
+import com.gu.scanamo.error.DynamoReadError
+import com.gu.scanamo.ops.{CatsInterpreter, ScanamoOps}
+import com.gu.scanamo.query.{Query, UniqueKey, UniqueKeys}
+import com.gu.scanamo.update.UpdateExpression
+
+object ScanamoCats {
+
+  def exec[F[_]: Effect, A](client: AmazonDynamoDBAsync)(op: ScanamoOps[A]): F[A] =
```

Comment:
This makes the `ScanamoAsync` implementation work for any types with an `Effect` instance, not just `IO`. That means `monix.eval.Task` and even `scalaz.ioeffect.IO` now have interoperability with this portion of the library. My aim wasn't just to support `IO`, but make it general enough to accommodate a wider range of Cats Effect users.

Reply:
Ooooh nice one! That gives us Monix, fs2 et al compat in one fell swoop then. While I was reviewing, I kept thinking that we now have these 5 interpreters with loads of boilerplate, and they are essentially all the same. It felt like `ScanamoCats` could be a nice template to abstract over any `M: Monad`, say 
```scala
trait Interpreter[M[_]] {
  type Client <: AmazonDynamoDB
  def exec[A](client: Client)(op: ScanamoOps[A])(implicit M: Monad[M]): F[A]
  
  def put[T: DynamoFormat](client: AmazonDynamoDBAsync)(tableName: String)(item: T)(implicit M: Monad[M]) : M[Option[Either[DynamoReadError, T]]] =
    exec(client)(ScanamoFree.put(tableName)(item))
  etc.
}
```
Then `Scanamo` would specialise to `Id`, `ScanamoScalaz` to `IO[E, ?]`, etc. Then I suppose it would break the existing contract, especially with the `Id` interpreter.

What do you think? It's something I'd be keen to investigate later.

Reply:
That would be a great improvement! I could implement something like that for the next PR